### PR TITLE
add metrec: arabic poetry dataset

### PR DESCRIPTION
**Pull request by zaidalyafeai**

### CODE REVIEW

File: datasets/metrec/metrec.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    def _generate_examples(self, directory, labeled=True):
+        """Generate examples."""
+        # For labeled examples, extract the label from the path.
+
+        with open(directory, encoding="UTF-8") as f:
+            for id_, record in enumerate(f.read().splitlines()):
+                label, bait = record.split(" ", 1)
+                print(label)
+                print(record.split(" ", 1))
```
Comment (lhoestq) [inline datasets/metrec/metrec.py:124]: Could you remove the print calls please ?

**Review by lhoestq — COMMENTED** [review]
That looks awesome good job !

For the sprint that is going to begin soon, we're now asking to add one dataset card per dataset.
You can se more information in those PRs : https://github.com/huggingface/datasets/pull/896 and https://github.com/huggingface/datasets/pull/894

Could you create a dataset card for Metrec as well ? The goal of a dataset card is to detail the  characteristics of a dataset. The content of the dataset card is going to be displayed in the huggingface.co dataset search tool :)

**Comment by zaidalyafeai** [comment]
@lhoestq removed prints and added the dataset card.

File: datasets/metrec/metrec.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        ]
+
+    def _generate_examples(self, directory, labeled=True):
+        """Generate examples."""
+        # For labeled examples, extract the label from the path.
+
+        with open(directory, encoding="UTF-8") as f:
+            for id_, record in enumerate(f.read().splitlines()):
+                label, bait = record.split(" ", 1)
+<<<<<<< HEAD
```
Comment (lhoestq) [inline datasets/metrec/metrec.py:130]: looks like you have some conflicts

File: datasets/metrec/README.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+---
```
Comment (lhoestq) [inline datasets/metrec/README.md:1]: really nice !

File: datasets/metrec/metrec.py (hunk 2 above)
Comment (zaidalyafeai) [reply datasets/metrec/metrec.py]: Oh shoot, fixed it.

**Comment by zaidalyafeai** [comment]
@lhoestq, I want to add other datasets as well. I am not sure if it is possible to do so with the same branch.

File: datasets/metrec/README.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+- unknown
+multilinguality:
+- monolingual
+size_categories:
+- 55K
+source_datasets:
+- original
+task_categories:
+- poetry_classification
+task_ids:
```
Comment (lhoestq) [inline datasets/metrec/README.md:18]:
```suggestion
task_categories:
- text-classification
task_ids:
- text-classification-other-poetry-classification
```

please use the tagger https://github.com/huggingface/datasets-tagging to see which categories are available

**Review by lhoestq — APPROVED** [review]
Really nice good job :) 

To add another dataset, please use a create a new branch from master and create another PR

**Comment by yjernite** [comment]
Hi @zaidalyafeai, really excited to get more Arabic coverage in the lib, thanks for your contribution!

Couple of last comments:
- this PR seems to modify some files that are unrelated to your dataset. Could you rebase from master? It should take care of that.
- The dataset card is a good start! Can you describe the task in a few words and add more information in the Data Structure part, including listing and describing the fields? Also, if you don't know how to fill out a paragraph, or if you have some information but think more would be beneficial, please leave `[More Information Needed]` instead of `[N/A]`

**Comment by zaidalyafeai** [comment]
@yjernite, I am not sure how to rebase. 

> Hi @zaidalyafeai, really excited to get more Arabic coverage in the lib, thanks for your contribution!
> 
> Couple of last comments:
> 
> * this PR seems to modify some files that are unrelated to your dataset. Could you rebase from master? It should take care of that.
> * The dataset card is a good start! Can you describe the task in a few words and add more information in the Data Structure part, including listing and describing the fields? Also, if you don't know how to fill out a paragraph, or if you have some information but think more would be beneficial, please leave `[More Information Needed]` instead of `[N/A]`

I have no idea how some other files changed. I tried to rebase and push but this created some errors. I had to run the command 
`git push -u --force origin add-metrec-dataset` which might cause some problems.

**Comment by lhoestq** [comment]
Feel free to create another branch/another PR without all the other changes

**Comment by zaidalyafeai** [comment]
@yjernite can you explain which other files are changed because of the PR ? https://github.com/huggingface/datasets/pull/893/files only shows files related to the dataset.

**Comment by lhoestq** [comment]
Right ! github is nice with us today :)

File: datasets/metrec/README.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+[N/A]
+
+### Licensing Information
+
+[N/A]
+
+### Citation Information
+
+[N/A]
```
Comment (lhoestq) [inline datasets/metrec/README.md:142]:
```suggestion
### Curation Rationale

[More Information Needed]

### Source Data

[More Information Needed]

#### Initial Data Collection and Normalization

The dataset was collected from [Aldiwan](https://www.aldiwan.net/).

#### Who are the source language producers?

The poems are from different poets.

### Annotations

The dataset does not contain any additional annotations.

#### Annotation process

[More Information Needed]

#### Who are the annotators?

[More Information Needed]

### Personal and Sensitive Information

[More Information Needed]

## Considerations for Using the Data

### Discussion of Social Impact and Biases

[More Information Needed]

### Other Known Limitations

[More Information Needed]

## Additional Information

### Dataset Curators

[More Information Needed]

### Licensing Information

[More Information Needed]

### Citation Information

[More Information Needed]
```

**Comment by lhoestq** [comment]
Looks like this one is ready to merge, thanks @zaidalyafeai !

**Comment by zaidalyafeai** [comment]
@lhoestq thanks for the merge. I am not a GitHub geek. I already have another dataset to add. I'm not sure how to add another given my forked repo. Do I  follow the same steps with a different checkout name ?

**Comment by lhoestq** [comment]
If you've followed the instructions in here : https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#start-by-preparing-your-environment

(especially point 2. and the command `git remote add upstream ....`)

Then you can try
```
git checkout master
git fetch upstream
git rebase upstream/master
git checkout -b add-<my-new-dataset-name>
```

**Pull request merged**